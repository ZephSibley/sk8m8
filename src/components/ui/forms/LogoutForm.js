import React from 'react';
import { Formik } from 'formik';
import Button from '@material-ui/core/Button';


const LogoutForm = ({ requests }) => {

    return (
        <div style={{position: "absolute", right: 0}}>
            <Formik
                
                onSubmit={(values, { setSubmitting }) => {
                    setSubmitting(true);
                    window.sessionStorage.removeItem('token')
                    window.location.href = "/"
                }}

                render={({
                    handleSubmit,
                    isSubmitting,
                }) => (
                        <form
                            onSubmit={handleSubmit}
                        >
                            <Button
                                color="primary"
                                component="button"
                                type="submit"
                                margin='normal'
                                disabled={isSubmitting}
                            >
                                Log out
                            </Button>
                        </form>
                    )}
            />
        </div>
    )
}

export default LogoutForm;