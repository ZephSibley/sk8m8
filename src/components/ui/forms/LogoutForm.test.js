import React from 'react';
import renderer from 'react-test-renderer';
import LogoutForm from './LogoutForm';

test('Basic Rendering', () => {
    const component = renderer.create(
        <LogoutForm />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})