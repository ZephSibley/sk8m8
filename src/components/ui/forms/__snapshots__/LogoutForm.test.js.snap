// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Rendering 1`] = `
<div
  style={
    Object {
      "position": "absolute",
      "right": 0,
    }
  }
>
  <form
    onSubmit={[Function]}
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      disabled={false}
      margin="normal"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Log out
      </span>
    </button>
  </form>
</div>
`;
